pkg/machine/keys.go

```diff
@@ -51,7 +52,22 @@ func CreateSSHKeysPrefix(dir string, file string, passThru bool, skipExisting bo
 // generatekeys creates an ed25519 set of keys
 func generatekeys(writeLocation string) error {
 	args := append(append([]string{}, sshCommand[1:]...), writeLocation)
-	return exec.Command(sshCommand[0], args...).Run()
+	cmd := exec.Command(sshCommand[0], args...)
+	stdErr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+	if err := cmd.Wait(); err == nil {
+		return nil
+	}
+	errMsg, err := io.ReadAll(stdErr)
+	if err != nil {
+		return err
+	}
+	return errors.Wrap(err, string(errMsg))
```

Comment:
This will not work, `ReadAll()` will overwrite `err` with `nil` so this function will return nil even when an error happend.
You have to store and return the `cmd.Wait()` error at the end.

also please use `fmt.Errorf("msg: %w",err)` for new code